File: 2015/01/javascript/main.mjs
import fs from "node:fs";

// --- Constants ---
const INPUT_FILE = "../input/input.txt";

// --- Parse the input file ---
const data = fs.readFileSync(INPUT_FILE, 'utf8');

// --- Puzzle 1 & 2 ---
var pos = 0;
var result = Array.from(data)
	.map((x, i) => [x, i])
	.reduce((acc, a) => {
		if (pos == 0 && acc == -1) pos = a[1];
		return acc + (a[0] == ')' ? -1 : 1);
	}, 0);

console.log("1. Floor: ", result.toLocaleString());
console.log("2. Position: ", pos.toLocaleString());

File: 2015/01/python/main.py
# --- Constants ---
INPUT_FILE = "../input/input.txt"

# --- Read input file ---
file = open(INPUT_FILE, mode='r')
data = file.read();
file.close()

# --- Puzzle 1 & 2 ---
floor = 0
pos = -1
for i, c in enumerate([*data]):
    if c == ')':
        floor -= 1
    else:
        floor += 1
    if pos == -1 and floor == -1:
        pos = i + 1;

# --- Results ---
print(f"1. Floor: {floor:,d}")
print(f"2. Position: {pos:,d}")
